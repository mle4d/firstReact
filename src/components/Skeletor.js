import React from 'react';

export default function Skeletor({ Name, Color, Trait, hex }) {
  const Skeletor = {
    Name,
    Color,
    Trait,
    hex
  };
  const fields = Object.keys(Skeletor)
    .map(key => {
      return (
        <>
          <dt>{key}</dt>
          <dd style={{ color: hex }}>{Skeletor[key]}</dd>
        </>
      );
    });
  return (
    <dl>
      {fields}
    </dl>
  );
}
