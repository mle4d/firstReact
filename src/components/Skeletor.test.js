import React from 'react';
import { shallow } from 'enzyme';
import Skeletor from './Skeletor';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Skeletor />);
    expect(wrapper).toMatchSnapshot();
  });
});
