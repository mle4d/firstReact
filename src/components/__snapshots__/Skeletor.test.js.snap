// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd
    style={
      Object {
        "color": undefined,
      }
    }
  />
  <dt>
    Color
  </dt>
  <dd
    style={
      Object {
        "color": undefined,
      }
    }
  />
  <dt>
    Trait
  </dt>
  <dd
    style={
      Object {
        "color": undefined,
      }
    }
  />
</dl>
`;
